src/Compilers/Core/Portable/DiagnosticAnalyzer/ShadowCopyAnalyzerAssemblyLoader.cs

```diff
@@ -212,5 +229,19 @@ private string CreateUniqueDirectoryForAssembly()
 
             return (directory, mutex);
         }
+
+        private static Guid ReadMvid(string filePath)
```

Comment:
Yes I simply copied this code from AssemblyUtilities.cs into this file. Unfortunately this file is source shared with the workspaces layer. At first I attempted to source share AssemblyUtilities.cs with workspaces too but that isn't enough because it pulls in extension methods from MetadataReaderExtensions.cs and that in turn brings in even more source files. Eventually just copied this 10 line method to make progress. 

I'm more than happy to hear suggestions on where to put this so it's accessible from both compiler and workspaces. 

Note: we should really just find a way to create an API that lets us have one copy of all this code.

Reply:
Is it possible to create a separate file for just this method and share that? Or at least add a comment to both locations saying that the code is shared.

Reply:
Ended up making `AssemblyUtilities` a `partial` type and moved the code that needed to be shared into the new file.